src/SprykerEco/Service/AkeneoPim/AkeneoPimService.php

```diff
@@ -0,0 +1,782 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Service\AkeneoPim;
+
+use Spryker\Service\Kernel\AbstractService;
+
+/**
+ * @method \SprykerEco\Service\AkeneoPim\AkeneoPimServiceFactory getFactory()
+ */
+class AkeneoPimService extends AbstractService implements AkeneoPimServiceInterface
+{
+    /**
+     * Specification:
+     *  - Gets a cursor to iterate over a list of products.
+     *
+     * @api
+     *
+     * @param int $pageSize The size of the page returned by the server.
+     * @param array $queryParameters Additional query parameters to pass in the request
+     *
+     * @return \SprykerEco\Service\AkeneoPim\Api\Wrapper\AkeneoResourceCursorInterface
+     */
+    public function getAllProducts($pageSize = 10, array $queryParameters = [])
+    {
+        return $this->getFactory()
+            ->createAkeneoPimAdapterFactory()
+            ->createProductApiAdapter()
+            ->all($pageSize, $queryParameters);
+    }
+
+    /**
+     * Specification:
+     *  - Gets a cursor to iterate over a list of categories.
+     *
+     * @api
+     *
+     * @param int $pageSize The size of the page returned by the server.
+     * @param array $queryParameters Additional query parameters to pass in the request
+     *
+     * @return \SprykerEco\Service\AkeneoPim\Api\Wrapper\AkeneoResourceCursorInterface
+     */
+    public function getAllCategories($pageSize = 10, array $queryParameters = [])
+    {
+        return $this->getFactory()
+            ->createAkeneoPimAdapterFactory()
+            ->createCategoryApiAdapter()
+            ->all($pageSize, $queryParameters);
+    }
+
+    /**
+     * Specification:
+     *  - Gets a cursor to iterate over a list of attributes.
+     *
+     * @api
+     *
+     * @param int $pageSize The size of the page returned by the server.
+     * @param array $queryParameters Additional query parameters to pass in the request
+     *
+     * @return \SprykerEco\Service\AkeneoPim\Api\Wrapper\AkeneoResourceCursorInterface
+     */
+    public function getAllAttributes($pageSize = 10, array $queryParameters = [])
+    {
+        return $this->getFactory()
+            ->createAkeneoPimAdapterFactory()
+            ->createAttributeApiAdapter()
+            ->all($pageSize, $queryParameters);
+    }
+
+    /**
+     * Specification:
+     *  - Gets a cursor to iterate over a list of attributes.
+     *
+     * @api
+     *
+     * @param string $attributeCode Code of the attribute
+     * @param int $pageSize The size of the page returned by the server.
+     * @param array $queryParameters Additional query parameters to pass in the request
+     *
+     * @return \Akeneo\Pim\ApiClient\Pagination\ResourceCursorInterface
```

Comment:
Please fix the return types

Reply:
What is wrong here?

Reply:
Service returns a 3rd party class, and actually it doesn't :)